### PR TITLE
cmake: fix cmake/googletest.cmake on Windows

### PR DESCRIPTION
**Pull request by apwojcik**
Windows file system is case insensitive. Both file names `GoogleTest.cmake` and `googletest.cmake` refer to the same file according to the file system.

### CODE REVIEW

File: test/gtest/CMakeLists.txt [hunk 1]
```diff
@@ -29,24 +23,25 @@ function(add_gtest TEST_NAME)
...
+add_gtest(conv_api)
+add_gtest(conv_api_strided_tensors)
+add_gtest(log_test)
+add_gtest(log_test_neg)
+add_gtest(na_infer)
+add_gtest(solver_convasm3x3u)
+
+if(NOT MIOPEN_BACKEND_OPENCL)
+  add_gtest(dumpTensorTest)
+endif()
```
Comment (JehandadKhan) [inline test/gtest/CMakeLists.txt]: I don't follow why the loop was removed, do you suggest for us to add each test explicitly ?

**Review by JehandadKhan — DISMISSED** [review]

File: test/gtest/CMakeLists.txt (hunk 1 above)
Comment (apwojcik) [reply test/gtest/CMakeLists.txt]: Yes. The `file(GLOBE...)` command was designed for CMake in _script mode_ (`*.cmake` files). Please read more about it here: https://cmake.org/cmake/help/latest/manual/cmake-language.7.html. The author's intention was not to use it in _build system generation mode_ - in this case. Please see the note for `GLOBE` here: https://cmake.org/cmake/help/latest/command/file.html#glob.
I will revert this change if you insist on returning the `file(GLOB...)` and the loop.

Comment (apwojcik) [reply test/gtest/CMakeLists.txt]: CMake scrip mode is not equivalent to `include(...)` files - that still counts as build system generation mode even when `.cmake` is used. CMake script mode is invoked with `cmake -P <script.cmake>` and does not configure anything and generate a build system.

Comment (averinevg) [reply test/gtest/CMakeLists.txt]: @apwojcik This is not a big problem in this case. I think these changes should be reverted

Comment (junliume) [reply test/gtest/CMakeLists.txt]: @apwojcik could you follow up with the review opinion above? Thanks!

Comment (apwojcik) [reply test/gtest/CMakeLists.txt]: done

**Review by junliume — APPROVED** [review]

**Pull request merged**